konrad/core.py

```diff
@@ -229,7 +231,15 @@ def is_converged(self):
             bool: ``True`` if converged, else ``False``.
         """
         # TODO: Implement proper convergence criterion (e.g. include TOA).
-        return np.all(np.abs(self.deltaT) < self.delta)
+        temp_test = np.all(np.abs(self.deltaT) < self.delta)
+        N_test = np.abs(self.deltaN) < self.delta2
+        if temp_test:
+            if N_test:
+                return True
+            else:
+                return False
+        else:
+            return False
```

Comment:
You can use boolean algebra
```suggestion
 return temp_test and N_test
```